tests/split_message_tests.py

```diff
@@ -0,0 +1,259 @@
+
+        first_call_args, _ = calls[0]
+        part_1 = first_call_args[2]
+        self.assertIn("message", part_1)
+        self.assertEqual(part_1["datapoints"], 1)
+        self.assertIn("MyDevice", part_1["message"])
+        self.assertEqual(part_1["message"]["MyDevice"], [{"temp": 22}])
+
+        second_call_args, _ = calls[1]
+        part_2 = second_call_args[2]
+        self.assertEqual(part_2["datapoints"], 1)
+        self.assertIn("MyDevice", part_2["message"])
+        self.assertEqual(part_2["message"]["MyDevice"], [{"humidity": 55}])
+
+        self.assertIsInstance(result, TBPublishInfo)
+
+    @patch('tb_device_mqtt.sleep', autospec=True)
+    @patch('tb_device_mqtt.log.warning', autospec=True)
+    def test_send_split_message_queue_size_retry(self, mock_log_warning, mock_sleep):
+        part = {'datapoints': 3, 'message': {"foo": "bar"}}
+        kwargs = {}
+        timeout = 10
+        device = "device2"
+        topic = "test/topic2"
+        msg_rate_limit = MagicMock()
+        dp_rate_limit = MagicMock()
+        msg_rate_limit.has_limit.return_value = True
+        dp_rate_limit.has_limit.return_value = True
+        self.client._wait_for_rate_limit_released = MagicMock(return_value=False)
+        self.client._client.publish.side_effect = [
+            self.fake_publish_queue, self.fake_publish_queue, self.fake_publish_ok
+        ]
+        self.client._TBDeviceMqttClient__error_logged = 0
+        with patch('tb_device_mqtt.monotonic', side_effect=[0, 12, 12, 12]):
+            results = []
+            ret = self.client._TBDeviceMqttClient__send_split_message(
+                results, part, kwargs, timeout, device, msg_rate_limit, dp_rate_limit, topic
+            )
+        self.assertEqual(self.client._client.publish.call_count, 3)
+        mock_log_warning.assert_called()
+        self.assertIsNone(ret)
+        self.assertIn(self.fake_publish_ok, results)
+
+class TestWaitUntilQueuedMessagesProcessed(unittest.TestCase):
+    @patch('tb_device_mqtt.sleep', autospec=True)
+    @patch('tb_device_mqtt.logging.getLogger', autospec=True)
+    @patch('tb_device_mqtt.monotonic')
+    def test_wait_until_current_queued_messages_processed_logging(self, mock_monotonic, mock_getLogger, mock_sleep):
+        client = TBDeviceMqttClient('fake_host', username="dummy_token", password="dummy")
+        fake_client = MagicMock()
+        fake_client._out_messages = [1, 2, 3, 4, 5, 6]
+        fake_client._max_inflight_messages = 5
+        client._client = fake_client
+        client.stopped = False
+        client.is_connected = MagicMock(return_value=True)
+        mock_monotonic.side_effect = [0, 6, 6, 1000]
+        fake_logger = MagicMock()
+        mock_getLogger.return_value = fake_logger
+        client._wait_until_current_queued_messages_processed()
+        fake_logger.debug.assert_called_with(
```

Comment:
Using a logger message as a result of the test is not correct.
A test can be affected by simply changing the logger message but tested functionality will still be correct.